macros/src/types/newtype.rs

```diff
@@ -27,7 +28,10 @@ pub(crate) fn newtype(attr: &StructAttr, name: &str, fields: &FieldsUnnamed) ->
         return super::unit::null(attr, name);
     }
 
-    let inner_ty = type_as.as_ref().unwrap_or(&inner.ty).clone();
+    let inner_ty = type_as
+        .as_ref()
+        .map(|ty_as| type_as_infer(ty_as, &inner.ty))
+        .unwrap_or_else(|| inner.ty.clone());
```

Comment:
Would it be possible to move this into a getter on `FieldAttr`, making the field `type_as` private?

Reply:
There's also the `format_variant` function that needs access to `type_as`